ceph_deploy/gatherkeys.py

```diff
@@ -1,84 +1,262 @@
 import os.path
 import logging
+import json
+import tempfile
+import shutil
+import time
 
 from ceph_deploy import hosts, exc
 from ceph_deploy.cliutil import priority
+from ceph_deploy.lib import remoto
 
 
 LOG = logging.getLogger(__name__)
 
 
-def fetch_file(args, frompath, topath, _hosts):
-    if os.path.exists(topath):
-        LOG.debug('Have %s', topath)
-        return True
-    else:
-        for hostname in _hosts:
-            filepath = frompath.format(hostname=hostname)
-            LOG.debug('Checking %s for %s', hostname, filepath)
-            distro = hosts.get(hostname, username=args.username)
-            key = distro.conn.remote_module.get_file(filepath)
+def _keyring_equivilent(keyring_one, keyring_two):
+    """
+    Check two keyrings are identical
+    """
+    def keyring_extract_key(file_path):
+        """
+        Cephx keyring files may or may not have white space before some lines.
+        They may have some values in quotes, so a safe way to compare is to
+        extract the key.
+        """
+        with file(file_path, 'r') as f:
+            for line in f:
+                content = line.strip()
+                if len(content) == 0:
+                    continue
+                split_line = content.split('=')
+                if split_line[0].strip() == 'key':
+                    return "=".join(split_line[1:]).strip()
+        raise RuntimeError("File '%s' is not a keyring" % file_path)
```

Comment:
The compare keyring could do with a check, but this nested function is nested because it is only to be used in the context of checking keyrings.